### PR TITLE
Promisified objects and added a single entry point createHypervisor()

### PR DESCRIPTION
**Pull request by atoy40**

### CODE REVIEW

**Comment by mbroadst** [comment]
@atoy40 this is strange, I think maybe you need to update your local fork since I added all of the exports stuff last night

**Comment by atoy40** [comment]
Argh I didn't see you did it !

**Comment by atoy40** [comment]
what do you thing about adding a link to the promise library into the libvirt javascript module ?
like :
```javascript
var Promise = require("bluebird");

.....

module.exports = {
  createHypervisor: function(uri, options) {
    return new libvirt.Hypervisor(uri, options);
  },
  promise: Promise,
};
```

**Comment by mbroadst** [comment]
@atoy40 I think that's a great idea, we've done that in sequelize as well. Generally its exported capitalized though.

File: src/network_filter.cc [hunk 1]
```diff
@@ -155,4 +155,3 @@ NLV_WORKER_EXECUTE(NetworkFilter, ToXml)
 }
 
 }   // namespce NodeLibvirt
-
```
Comment (mbroadst) [inline src/network_filter.cc]: removing every final line :) this is usually a requirement, I'm surprised github isn't complaining about it.

**Comment by atoy40** [comment]
you mean Promise ? or PROMISE ?

**Comment by mbroadst** [comment]
"Promise"

**Comment by mbroadst** [comment]
merged with 451a03445c4026ec9b1360f1ec8dd3831d9c1993

**Pull request closed without merging**